CMakeLists.txt

```diff
@@ -0,0 +1,341 @@
+torch/python via pybind and is required by Aphrodite extensions for this
+reason. So, add it by manually using `append_torchlib_if_found` from
+torch's cmake setup.
+#]==============================================================================]
+append_torchlib_if_found(torch_python)
+
+#[====================================================================[
+# Set up GPU language and check the torch version and warn if it isn't
+# what is expected.
+#]====================================================================]
+if (NOT HIP_FOUND AND CUDA_FOUND)
+  set(APHRODITE_GPU_LANG "CUDA")
+
+  if (NOT Torch_VERSION VERSION_EQUAL ${TORCH_SUPPORTED_VERSION_CUDA})
+    message(WARNING "Pytorch version ${TORCH_SUPPORTED_VERSION_CUDA} "
+      "expected for CUDA build, saw ${Torch_VERSION} instead.")
+  endif()
+elseif(HIP_FOUND)
+  set(APHRODITE_GPU_LANG "HIP")
+
+  #[===========================================================================[
+  Importing torch recognizes and sets up some HIP/ROCm configuration but does
+  not let cmake recognize .hip files. In order to get cmake to understand the
+  .hip extension automatically, HIP must be enabled explicitly.
+  #]===========================================================================]
+  enable_language(HIP)
+
+  # ROCm 5.x
+  if (ROCM_VERSION_DEV_MAJOR EQUAL 5 AND
+      NOT Torch_VERSION VERSION_EQUAL ${TORCH_SUPPORTED_VERSION_ROCM_5X})
+    message(WARNING "Pytorch version ${TORCH_SUPPORTED_VERSION_ROCM_5X} "
+      "expected for ROCMm 5.x build, saw ${Torch_VERSION} instead.")
+  endif()
+
+  # ROCm 6.x
+  if (ROCM_VERSION_DEV_MAJOR EQUAL 6 AND
+      NOT Torch_VERSION VERSION_EQUAL ${TORCH_SUPPORTED_VERSION_ROCM_6X})
+    message(WARNING "Pytorch version ${TORCH_SUPPORTED_VERSION_ROCM_6X} "
+      "expected for ROCMm 6.x build, saw ${Torch_VERSION} instead.")
+  endif()
+else()
+  message(FATAL_ERROR "Can't find CUDA or HIP installation.")
+endif()
+
+#[=========================================================================[
+Override the GPU architectures detected by cmake/torch and filter them by
+the supported versions for the current language.
+The final set of arches is stored in `APHRODITE_GPU_ARCHES`.
+#]=========================================================================]
+override_gpu_arches(APHRODITE_GPU_ARCHES
+  ${APHRODITE_GPU_LANG}
+  "${${APHRODITE_GPU_LANG}_SUPPORTED_ARCHS}")
+
+#[=========================================================================[
+Query torch for additional GPU compilation flags for the given
+`APHRODITE_GPU_LANG`.
+The final set of arches is stored in `APHRODITE_GPU_FLAGS`.
+#]=========================================================================]
+get_torch_gpu_compiler_flags(APHRODITE_GPU_FLAGS ${APHRODITE_GPU_LANG})
+
+# Set nvcc parallelism.
+if(NVCC_THREADS AND APHRODITE_GPU_LANG STREQUAL "CUDA")
+  list(APPEND APHRODITE_GPU_FLAGS "--threads=${NVCC_THREADS}")
+endif()
```

Comment:
See comment in setup.py about nvcc --threads seemingly not doing anything, at least with the Ninja backend with CMake